src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -343,17 +363,63 @@ localCompletionsForParsedModule pm@ParsedModule{pm_parsed_source = L _ HsModule{
 
     thisModName = ppr hsmodName
 
+    --recordCompls = localRecordSnippetProducer pm thisModName
+
+findRecordCompl :: ParsedModule -> T.Text -> Located (IdP GhcPs) -> HsDataDefn GhcPs -> [CompItem]
+findRecordCompl pmod mn lname dd = name_type''
+    where
+        conDecl = unLoc <$> dd_cons dd
+        h98 = catMaybes $ findH98 <$> conDecl
+        findH98 conDecl = case conDecl of
+          ConDeclH98{..} -> Just (unLoc con_name, con_args)
+          ConDeclGADT{} -> Nothing  -- TODO: Expand this out later
+          _ -> Nothing
+
+        name_flds :: [(RdrName, [ConDeclField GhcPs])]
+        name_flds = catMaybes $ decompose <$> h98
+
+        decompose x = case getFlds $ (snd x) of
+                          Just con_details -> Just (fst x, con_details)
+                          Nothing -> Nothing
+
+        getFlds :: HsConDetails arg (Located [LConDeclField GhcPs]) -> Maybe [SrcSpanLess (LConDeclField GhcPs)]
+        getFlds conArg = case conArg of
+                             RecCon rec -> Just $ unLoc <$> (unLoc rec)
+                             _ -> Nothing
+
+        --name_type :: [(RdrName, (Located RdrName, HsType GhcPs))]
+
+        name_type = decompose' <$> name_flds
+        decompose' x = (fst x, catMaybes $ extract <$> (snd x))
+
+        extract ConDeclField{..} = let
+            fld_type = unLoc cd_fld_type
+            fld_name = rdrNameFieldOcc $ unLoc . head $ cd_fld_names --TODO: Why is cd_fld_names a list?
+            in
+                Just (fld_name, fld_type)
+        extract _ = Nothing
+
+        name_type'' = decompose'' <$> name_type
+        decompose'' :: (RdrName, [(Located RdrName, HsType GhcPs)]) -> CompItem
+        decompose'' x = let
+            ctxStr = T.pack . showGhc . fst $ x
+            flds = (\(x', y') -> (T.pack . showGhc . unLoc $ x', T.pack . showGhc $ y'))  <$> (snd x)
+            doc = SpanDocText (getDocumentation [pmod] lname) (SpanDocUris Nothing Nothing)
+            in
+                mkRecordSnippetCompItem ctxStr flds mn doc
+
+
 ppr :: Outputable a => a -> T.Text
 ppr = T.pack . prettyPrint
 
 newtype WithSnippets = WithSnippets Bool
 
 toggleSnippets :: ClientCapabilities -> WithSnippets -> CompletionItem -> CompletionItem
 toggleSnippets ClientCapabilities { _textDocument } (WithSnippets with) x
-  | with && supported = x
+  | with || supported = x
```

Comment:
Note that I had to make this change to let snippets show up. I think we need to make a more relevant change to the code https://github.com/haskell/ghcide/blob/dcf680402722e7660377bf7da3fca27c2bde988f/src/Development/IDE/Plugin/Completions.hs#L134-L135